engine/src/collisions/narrow_phase/plugin.cpp

```diff
@@ -65,4 +66,86 @@ void cubos::engine::narrowPhaseCollisionsPlugin(Cubos& cubos)
+
+                        if (intersects)
+                        {
+                            cmds.relate(ent1, ent2,
+                                        CollidingWith{.entity = ent1,
+                                                      .entity1OriginalPosition = position1.vec,
+                                                      .entity2OriginalPosition = position2.vec,
+                                                      .penetration = intersectionInfo.penetration,
+                                                      .position = {0.0F, 0.0F, 0.0F},
+                                                      .normal = intersectionInfo.normal});
+                            break;
+                        }
+                    }
+                }
+            });
+
+    cubos.system("find colliding voxel-voxel pairs")
+        .tagged(collisionsNarrowTag)
+        .after(collisionsBroadTag)
+        .call([](Commands cmds, Query<Entity, const Position&, const LocalToWorld&, const VoxelCollisionShape&,
+                                      PotentiallyCollidingWith&, Entity, const Position&, const LocalToWorld&,
+                                      const VoxelCollisionShape&>
+                                    query) {
+            for (auto [ent1, position1, localToWorld1, voxelShape1, potentiallyCollidingWith, ent2, position2,
+                       localToWorld2, voxelShape2] : query)
+            {
+                cubos::core::geom::Intersection intersectionInfo{};
+                intersectionInfo.normal = glm::vec3({0.0F});
+                intersectionInfo.penetration = std::numeric_limits<float>::infinity();
+
+                for (const auto box1 : voxelShape1.getBoxes())
+                {
+                    auto pos1 = localToWorld1.mat;
+                    pos1[3] += glm::vec4(box1.shift, 0.0f);
+                    bool intersects = false;
+                    for (const auto box2 : voxelShape2.getBoxes())
+                    {
+                        auto pos2 = localToWorld1.mat;
+                        pos2[3] += glm::vec4(box2.shift, 0.0f);
+                        intersects = cubos::core::geom::intersects(box1.box, pos1, box2.box, pos2, intersectionInfo);
+
+                        if (intersects)
+                        {
+                            cmds.relate(ent1, ent2,
+                                        CollidingWith{.entity = ent1,
+                                                      .entity1OriginalPosition = position1.vec,
+                                                      .entity2OriginalPosition = position2.vec,
+                                                      .penetration = intersectionInfo.penetration,
+                                                      .position = {0.0F, 0.0F, 0.0F},
+                                                      .normal = intersectionInfo.normal});
+                            break;
```

Comment:
This is still missing the multiple intersection handling code. Maybe we should leave it to another PR, what do you think?